PokeAlarm/Manager.py

```diff
@@ -110,11 +109,11 @@ def stop(self):
         self.__event.set()
 
     def join(self):
-        self.__process.join(timeout=10)
-        if self.__process.is_alive():
+        self.__process.join(timeout=20)
+        if not self.__process.ready():
             log.warning("Manager {} could not be stopped in time!"
-                        + " Forcing process to stop.")
-            self.__process.terminate()
+                        " Forcing process to stop.".format(self.__name))
+            self.__process.kill(timeout=2, block=True)  # Force stop
         else:
             log.info("Manager {} successfully stopped!".format(self.__name))
```

Comment:
In normal closing, it does not appear that the system gets to this point before the process exits.